### PR TITLE
Backport to 8.x #2604 Allow delete of single value in a config object

### PR DESCRIPTION
**Pull request by AdamPS**
@weitzman Thanks for the original fix. I hope you will accept this back-port to 8.x.

### CODE REVIEW

File: commands/core/config.drush.inc [hunk 1]
```diff
@@ -267,14 +268,24 @@ function drush_config_get($config_name, $key = NULL) {
...
+function drush_config_delete($config_name, $key = null) {
   $config =\Drupal::service('config.factory')->getEditable($config_name);
   if ($config->isNew()) {
     return drush_set_error('DRUSH_CONFIG_ERROR', 'Configuration name not recognized. Use config-list to see all names.');
   }
   else {
-    $config->delete();
+    if ($key) {
+      if ($config->get($key) === null) {
+        throw new \Exception(dt('Configuration key !key not found.', array('!key' => $key)));
```
Comment (weitzman) [inline commands/core/config.drush.inc]: should use `return drush_set_error(...key not found...);` in drush8

**Comment by AdamPS** [comment]
@weitzman Thanks for the review - done.  I also fixed a missing dt().

**Comment by weitzman** [comment]
Thanks

**Comment by AdamPS** [comment]
Many thanks

**Pull request merged**